Toan nguyen26 js wip

Toan-Nguyen26 opened the pull request:
Started to mov my js to Chatbase , still under progress

Toan-Nguyen26 committed:
(WIP) Added my own myAjaxSubmit 

A lot of things might be wrong and broken but it seems fine for now

Toan-Nguyen26 committed:
WIP 

I think I commit to a wrong file last time , here is the newer version which is also WIP

Toan-Nguyen26 committed:
(WIP) Added my own js

Toan-Nguyen26 committed:
(WIP) moving my js to chatbase

Toan-Nguyen26 committed:
(WIP) some more js conversion

Toan-Nguyen26 committed:
(WIP) Chatbase function converted to js

Toan-Nguyen26 commented:
More function was converted

Toan-Nguyen26 committed:
My own promise added (WIP)

Toan-Nguyen26 commented:
The promise was kinda done, not the best way to check for document ready but it's what I came up with

ericras commented on www/js/VisitorChat/5.0/ChatBase.js.php:
```diff
@@ -97,7 +97,8 @@
 
         if (typeof visitorchat_config == "object") {
-            $.extend(this.config, visitorchat_config);
+            extend(this.config, visitorchat_config);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign

ericras commented on www/js/VisitorChat/5.0/ChatBase.js.php:
```diff
@@ -574,7 +619,7 @@
         //$.parseHTML(html)[0].outerHTML is used to compare the rendered html (the browser can change quotes, etc)
         //It just makes the comparison more accurate (at the cost of a little speed)
-        if ($(selector).html() === $.parseHTML(html)[0].outerHTML) {
+        if (selector.innerHTML === $.parseHTML((html)[0].outerHTML)) {
```
"selector" by itself probably doesn't work.  Do you mean "document.querySelector(selector)" ?

ericras commented on www/js/VisitorChat/5.0/ChatBase.js.php:
```diff
@@ -631,42 +686,51 @@
+                // submit() and val('') is a jquery function imo
                 if (VisitorChat.chatStatus == 'LOGIN') {
-                  $('#visitorchat_clientLogin').submit();
+                   document.querySelector('visitorchat_clientLogin').submit();
```
Missing the # on the id: #visitorchat_clientLogin

Toan-Nguyen26 committed:
Clean/Update chatbase code

Toan-Nguyen26 committed:
Slight update (WIP)

Toan-Nguyen26 committed:
Overhaul on a lot of jQuery (WIP)

Toan-Nguyen26 committed:
More JS conversion done (WIP)

Toan-Nguyen26 committed:
(WIP) Update Client.js.php

Toan-Nguyen26 committed:
(WIP) Update Chatbase

Toan-Nguyen26 committed:
(WIP) Added some comments for clarification

Toan-Nguyen26 committed:
(WIP) more js conversion

Toan-Nguyen26 committed:
(WIP) more js conversion

Toan-Nguyen26 committed:
(WIP) some ajax fixing + js conversion

Toan-Nguyen26 committed:
(WIP) js conversion

Toan-Nguyen26 committed:
(WIP) some slight change

A bug was found when the operator denied the operator message , the chat box is froze after that

Toan-Nguyen26 committed:
(WIP)

Toan-Nguyen26 committed:
(WIP) Chatbase only have ajax func

Toan-Nguyen26 committed:
(WIP) ajax , unbind and some other func

Toan-Nguyen26 commented:
Most of them have ajax func and some other that are still jquery
